From 4afd9d637078044d3308a1a8590a1704e7b2eed8 Mon Sep 17 00:00:00 2001
From: Pierre Couillaud <pierre@broadcom.com>
Date: Fri, 20 Mar 2026 07:56:18 -0700
Subject: [PATCH 147/223] Reapply "UPSTREAM: PCI/pwrctrl: Create pwrctrl
 devices only when CONFIG_PCI_PWRCTRL is enabled"

This reverts commit 490ce5e6fdf834e30647135042aaa0350c8ded0c.

Bug: 513283531
Bug: 494543965
Bug: 513342406
Change-Id: Ib6bc13604cbc3ead07414ec8e018b81828968d74
Signed-off-by: Pierre Couillaud <pierre@broadcom.com>
(cherry picked from commit 44ff504b79e3fdb96eaf013a13e387baeac3be7c)
---
 drivers/pci/probe.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/drivers/pci/probe.c b/drivers/pci/probe.c
index 1b080e4e423f..61d2f12996d3 100644
--- a/drivers/pci/probe.c
+++ b/drivers/pci/probe.c
@@ -2455,6 +2455,7 @@ bool pci_bus_read_dev_vendor_id(struct pci_bus *bus, int devfn, u32 *l,
 }
 EXPORT_SYMBOL(pci_bus_read_dev_vendor_id);
 
+#if IS_ENABLED(CONFIG_PCI_PWRCTRL)
 static struct platform_device *pci_pwrctrl_create_device(struct pci_bus *bus, int devfn)
 {
 	struct pci_host_bridge *host = pci_find_host_bridge(bus);
@@ -2497,6 +2498,12 @@ static struct platform_device *pci_pwrctrl_create_device(struct pci_bus *bus, in
 
 	return NULL;
 }
+#else
+static struct platform_device *pci_pwrctrl_create_device(struct pci_bus *bus, int devfn)
+{
+	return NULL;
+}
+#endif
 
 /*
  * Read the config data for a PCI device, sanity-check it,
-- 
2.50.1

